airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/StagingConsumerFactory.java

```diff
@@ -160,15 +167,32 @@ private static AirbyteStreamNameNamespacePair toNameNamespacePair(final WriteCon
    * @param catalog collection of configured streams (e.g. API endpoints or database tables)
    * @return
    */
-  private CheckedBiConsumer<AirbyteStreamNameNamespacePair, SerializableBuffer, Exception> flushBufferFunction(
+  @VisibleForTesting
+  CheckedBiConsumer<AirbyteStreamNameNamespacePair, SerializableBuffer, Exception> flushBufferFunction(
                                                                                                                final JdbcDatabase database,
                                                                                                                final StagingOperations stagingOperations,
                                                                                                                final List<WriteConfig> writeConfigs,
                                                                                                                final ConfiguredAirbyteCatalog catalog) {
-    final Map<AirbyteStreamNameNamespacePair, WriteConfig> pairToWriteConfig =
-        writeConfigs.stream()
-            .collect(Collectors.toUnmodifiableMap(
-                StagingConsumerFactory::toNameNamespacePair, Function.identity()));
+    final Set<WriteConfig> conflictingStreams = new HashSet<>();
```

Comment:
This block seems to make sense in the `onStartFunction` since none of the values you're accessing here seem to be only isolated to the `flushBufferFunction` section

The reason that this makes sense to me in the `onStartFunction` is primarily because you already have the writeConfig values (since we're setting up the temporary tables for each stream (synonymous with writeConfig) and the metadata for each `streamIdentifier` will be known at setup time

It also fits with what `onStartFunction` means which is set up so this is only called _once_ as opposed to many times throughout the flushing of the buffer (N times where N can be infinitely large)

Reply:
this block happens before the lambda gets created (i.e. it's not actually part of the flush buffer function), so I think it only gets executed once per sync

Reply:
That does make sense, although I would be hesitant to have this in the `flushBufferFunction` since it muddies the singular responsibility of the buffer flush. Also, since we know that the lambda for `onStartFunction` is called only once and it already includes `writeConfigs` that it would be another suitable location

That said, this isn't blocking